app/models/concerns/project_media_creators.rb

```diff
@@ -263,6 +263,6 @@ def create_claim_description_and_fact_check
   end
 
   def create_tags
-    self.set_tags.each { |tag| Tag.create!(annotated: self, tag: tag.strip, skip_check_ability: true) } if self.set_tags.is_a?(Array)
+    self.set_tags.each { |tag| GenericWorker.perform_in(1.second, 'Tag', 'create!', annotated_type: 'ProjectMedia' , annotated_id: self.id, tag: tag.strip, skip_check_ability: true) } if self.set_tags.is_a?(Array)
```

Comment:
Suggestion: I think it's going to be easier to encapsulate and to unit-test if we wrap the whole execution in a single worker run. Something like:

```ruby
def create_tags
  if self.set_tags.is_a?(Array)
    GenericWorker.perform_in(1.second, 'ProjectMedia', 'create_tags_in_background', self.id, self.set_tags.to_json)
  end
end

def self.create_tags_in_background(project_media_id, tags_array_as_json)
  project_media = ProjectMedia.find(project_media_id)
  tags = JSON.parse(tags_array_as_json)
  tags.each { |tag| Tag.create! annotated: project_media, tag: tag.strip, skip_check_ability: true) }
end
```

Rationale:

- For clarity, I think it's important for the signature of `GenericWorker` to be explicitly class/method/arguments (I see it's implicit now) - more on that in the next comment
- Each tag in a separate background job run means background job overhead for a small task, and it can also lead to inconsistencies